lms/djangoapps/grades/new/course_grade.py

```diff
@@ -123,26 +120,42 @@ def compute_and_update(self, read_only=False):
 
         If read_only is True, doesn't save any updates to the grades.
         """
-        self._log_event(log.warning, u"compute_and_update, read_only: {}".format(read_only))
         subsection_grade_factory = SubsectionGradeFactory(self.student, self.course, self.course_structure)
+        subsections_total = 0
+        blocks_total = 0
         for chapter_key in self.course_structure.get_children(self.course.location):
             chapter = self.course_structure[chapter_key]
             chapter_subsection_grades = []
             for subsection_key in self.course_structure.get_children(chapter_key):
-                chapter_subsection_grades.append(
-                    subsection_grade_factory.create(self.course_structure[subsection_key], read_only=True)
+                subsections_total = subsections_total + 1
+                subsection_grade = subsection_grade_factory.create(
+                    self.course_structure[subsection_key],
+                    read_only=True
                 )
+                chapter_subsection_grades.append(subsection_grade)
+                blocks_total = blocks_total + len(subsection_grade.locations_to_weighted_scores)
```

Comment:
Nit: blocks_total can just be `len(self.locations_to_weighted_scores)` after the `for` loop.